threads.go

```diff
@@ -555,42 +578,51 @@ func (t *threads) putRecord(ctx context.Context, id thread.ID, lid peer.ID, rec
 		return err
 	}
 
-	// Save the record locally
-	// Note: These get methods will return cached nodes.
-	block, err := rec.GetBlock(ctx, t)
-	if err != nil {
-		return err
-	}
-	event, ok := block.(*cbor.Event)
-	if !ok {
-		return fmt.Errorf("invalid event")
-	}
-	header, err := event.GetHeader(ctx, t, nil)
-	if err != nil {
-		return err
-	}
-	body, err := event.GetBody(ctx, t, nil)
-	if err != nil {
-		return err
-	}
-	err = t.AddMany(ctx, []format.Node{rec, event, header, body})
-	if err != nil {
-		return err
-	}
+	for i := len(unknownRecords) - 1; i >= 0; i-- {
```

Comment:
Iterate from right to left (so, in inverse log order) each unknown record.
This will process things _in order_, in particular, will notify in `bus` in the correct order.

Reply:
Sweet! For extra large traversals, say `unknownRecords` becomes _massive_, we could write these to disk as an unprocessed record queue.

Reply:
Yes, that's right... will grow linearly with how far behind we're. 
That sound a reasonable option, we can think it further.